### PR TITLE
chore: static site copy should use correct content-type

### PR DESCRIPTION
**Pull request by iamhopaul123**
An alternative way solving content-type issue instead of https://github.com/aws/copilot-cli/pull/4857. Resolving the possible edge case when two identical files use different extensions.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: cf-custom-resources/lib/copy-assets.js [hunk 1]
```diff
@@ -9,10 +9,19 @@ const aws = require("aws-sdk");
...
-	}).promise();
-};
\ No newline at end of file
+  const s3 = new aws.S3();
+  await s3
+    .copyObject({
+      CopySource: event.srcBucket + "/" + event.mapping.path,
+      Bucket: event.destBucket,
+      Key: event.mapping.destPath,
+      ContentType:
```
Comment (dannyrandall) [inline cf-custom-resources/lib/copy-assets.js:18]: what happens if this is set to `""`?

Comment (iamhopaul123) [reply cf-custom-resources/lib/copy-assets.js:18]: no error but doesn't make any sense either.

**Review by dannyrandall — APPROVED** [review]
nice!!

**Review by Lou1415926 — APPROVED** [review]

File: internal/pkg/deploy/upload/asset/asset.go [hunk 2]
```diff
@@ -43,6 +44,7 @@ type asset struct {
 
 	ArtifactBucketPath string `json:"path"`
 	ServiceBucketPath  string `json:"destPath"`
+	ContentType        string `json:"contentType"`
```
Comment (efekarakus) [inline internal/pkg/deploy/upload/asset/asset.go:47]: nit: do you think it's worthwhile doing `,omitempty` for these tags to just have a file that's slightly smaller if the field is empty?

File: internal/pkg/deploy/upload/asset/asset.go [hunk 3]
```diff
@@ -180,12 +183,12 @@ func (u *ArtifactBucketUploader) uploadAssetMappingFile(assets []asset) (string,
 
 // dedupe returns a copy of assets with duplicate entries removed.
 func dedupe(assets []asset) []asset {
-	type key struct{ a, b string }
+	type key struct{ a, b, c string }
```
Comment (efekarakus) [inline internal/pkg/deploy/upload/asset/asset.go:186]: nit:
```go
type key struct{ field1, field2, field3 string }
```

File: internal/pkg/deploy/upload/asset/asset.go (hunk 2 above)
Comment (iamhopaul123) [reply internal/pkg/deploy/upload/asset/asset.go:47]: I thought about this but the file itself is relatively small and maybe it's good to keep each element in the array consistent to have the same key set?

File: internal/pkg/deploy/upload/asset/asset.go (hunk 3 above)
Comment (iamhopaul123) [reply internal/pkg/deploy/upload/asset/asset.go:186]: Added in https://github.com/aws/copilot-cli/pull/4870

**Pull request merged**